### PR TITLE
Component style updates + for redesign 

### PR DESCRIPTION
**Pull request by Erin007**
A handful of changes are happening here: 

1.) teacherHomepage folder was renamed to studioHomepages because the folder contains components not only for the teacher homepage, but also for the student homepage and /courses 

2.) New React components now have sharp rather than rounded corners. 

   Example CourseCard before: 
rounded corners - before

   Example CourseCard after: 
sharp corners - after

3.) The purple user_hero now appears on /courses for students (it's currently on /home for students but since /home redirects to /courses for students for now, it also needs to be on /courses). user_hero now has sharp corners and styling of View Courses button changed to (hopefully) indicate  it is the secondary call to action. 

updated user hero

4.) Students without course progress will no longer see a Recent Courses section; this is accomplished by passing isTeacher to RecentCourseCollapsible in the Courses component and conditionally rendering content specific to the user and their course progress. Previously students without courses were seeing a Set Up Message directed to teachers, which didn't make sense. 

Student without courses on /courses before: 
student on courses without courses before

Student without courses on /courses after: 
student no courses after

### CODE REVIEW

**Comment by Erin007** [comment]
cc: @tanyaparker @joshlory

**Comment by Erin007** [comment]
Totally realized as I was writing the description of all of the changes, that this is beast PR that should actually be 4 separate PRs. I'm going to tackle the merge conflicts from renaming that file first; then, let me know if I need to divide out the changes in more digestible chunks.

File: apps/src/templates/studioHomepages/RecentCoursesCollapsible.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import React, { PropTypes } from 'react';
+import CollapsibleSection from './CollapsibleSection';
+import CourseCard from './CourseCard';
+import SetUpMessage from './SetUpMessage';
+import i18n from "@cdo/locale";
+
+const RecentCoursesCollapsible = React.createClass({
+  propTypes: {
+    courses: PropTypes.arrayOf(
```
Comment (breville) [inline apps/src/templates/studioHomepages/RecentCoursesCollapsible.jsx]: can we use the shape defined in shapes.js?

File: dashboard/app/views/courses/index.html.haml [hunk 2]
```diff
@@ -23,5 +23,9 @@
   = render file: Rails.root.join('..', 'shared', 'haml', 'course_explorer_table.haml')
   = render file: Rails.root.join('..', 'shared', 'haml', 'tool_explorer_table.haml')
   = render partial: 'home/all_courses'
+  -if current_user
+    - unless current_user.teacher?
+      #user_hero
+        = render template: 'api/user_hero'
```
Comment (breville) [inline dashboard/app/views/courses/index.html.haml]: Is it bad if we skip these checks to reduce the amount of ruby code, and just rely on the React code to show it or not?

Comment (breville) [reply dashboard/app/views/courses/index.html.haml]: That said, if we do have to leave in the ruby code, we could make it one line with `if current_user && !current_user.teacher?`

File: dashboard/app/views/home/_homepage.html.haml [hunk 3]
```diff
@@ -8,9 +8,10 @@
 
 -# Components generated by HAML which might be pulled in and displayed by the React model.
 .hiddenComponents{style: "display: none"}
-  - unless current_user.teacher?
-    #user_hero
-      = render template: 'api/user_hero'
+  -if current_user
```
Comment (breville) [inline dashboard/app/views/home/_homepage.html.haml]: similar comments here on ruby

File: apps/src/templates/studioHomepages/RecentCoursesCollapsible.jsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+          <CollapsibleSection
+            header={i18n.courses()}
+            linkText={i18n.viewAllCourses()}
+            link="/courses"
+            showLink={showAllCoursesLink}
+          >
+            {courses.map((course, index) =>
+              <CourseCard
+                key={index}
+                courseName={course.courseName}
```
Comment (Bjvanminnen) [inline apps/src/templates/studioHomepages/RecentCoursesCollapsible.jsx]: Probably the merge conflicts you mentioned, but this should now be `course.name`

**Pull request merged**